prometheus-client/docs/memory-use.md

```diff
@@ -0,0 +1,91 @@
+# Investigating Memory Use in `prometheus-client`
```

Comment:
This feels like an odd file to check into main. I think this would be more appropriate as the PR description.